src/hotspot/share/opto/superword.cpp

```diff
@@ -1001,6 +1005,17 @@ bool SuperWord::isomorphic(Node* s1, Node* s2) {
   }
 }
 
+// Do we have pattern n1 = (iv + c) and n2 = (iv + c + 1)?
```

Comment:
Why  `+1` is special case?  Does SuperWord handle case when loop's increment is not `1`?

Reply:
Yes, currently it only seems to handle positive `1`.

Relevant line in `SuperWord::vector_opd`, the only place where the `PopulateIndexNode` is created:
`Node* vn = new PopulateIndexNode(iv(), igvn().intcon(1), vt);`

The verification code is at the beginning of `SuperWord::is_vector_use`:
`offset_t->get_con() != (jint) i` -> we expect `populate_index->at(i) = (iv + i)`

That is a limitation we can surely remove in a future RFE.

Reply:
I finally got what this function for :^)  Looks reasonable.

Reply:
Let me know if you have a better idea to make it more clear then ;)

Reply:
Adjust comment:  "Look for pattern ... which may lead to PopulateIndex vector node creation so we can skip it"